codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -243,8 +243,9 @@ private void addOperationDocumentShapeBindersForSerializer(GenerationContext con
 
     private void generateOperationSerializerMiddleware(GenerationContext context, OperationShape operation) {
         GoStackStepMiddlewareGenerator middleware = GoStackStepMiddlewareGenerator.createSerializeStepMiddleware(
-                ProtocolGenerator.getSerializeMiddlewareName(operation.getId(), getProtocolName()
-                ));
+                ProtocolGenerator.getSerializeMiddlewareName(operation.getId(), getProtocolName()),
+                "OperationSerializer"
```

Comment:
Why not just have the functions set these so we don't have to populate this everywhere? If nothing else, these should probably be public static strings somewhere.

Reply:
yeah can move this to a more generic method for getting the generator for the operation serializer middleware

Reply:
Ended up moving these to constants. We're holding off publishing these as constants within the Go code until we have a better understanding of user story for helpers that will be created for adding and removing specific middleware. Middleware IDs like these might end up being specified as consts in the SDK's core library.